src/sentry/api/endpoints/organization_member_invite/reinvite.py

```diff
@@ -0,0 +1,150 @@
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.organizationmemberinvite import (
+    OrganizationMemberReinviteRequestValidator,
+)
+from sentry.models.organization import Organization
+from sentry.models.organizationmemberinvite import OrganizationMemberInvite
+from sentry.utils import metrics
+
+ERR_EXPIRED = "You cannot resend an expired invitation without regenerating the token."
+ERR_INSUFFICIENT_SCOPE = "You are missing the member:admin scope."
+ERR_INVITE_UNAPPROVED = "You cannot resend an invitation that has not been approved."
+ERR_MEMBER_INVITE = "You cannot resend an invitation that was sent by someone else."
+ERR_RATE_LIMITED = "You are being rate limited for too many invitations."
+
+
+class MemberReinvitePermission(OrganizationPermission):
+    scope_map = {"PUT": ["member:invite", "member:write", "member:admin"]}
+
+
+@region_silo_endpoint
+class OrganizationMemberReinviteEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "PUT": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ENTERPRISE
+    permission_classes = (MemberReinvitePermission,)
+
+    def convert_args(
+        self,
+        request: Request,
+        member_invite_id: str,
+        organization_id_or_slug: str | int | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> tuple[tuple[Any, ...], dict[str, Any]]:
+        args, kwargs = super().convert_args(request, organization_id_or_slug, *args, **kwargs)
+
+        try:
+            kwargs["invited_member"] = OrganizationMemberInvite.objects.get(
+                id=int(member_invite_id)
+            )
+        except OrganizationMemberInvite.DoesNotExist:
+            raise ResourceDoesNotExist
+        return args, kwargs
+
+    def _reinvite_and_send_email(
+        self,
+        request: Request,
+        organization: Organization,
+        invited_member: OrganizationMemberInvite,
+        trigger_regenerate_token: bool,
+    ) -> Response:
+        if not invited_member.invite_approved:
+            return Response({"detail": ERR_INVITE_UNAPPROVED}, status=400)
+
+        if ratelimits.for_organization_member_invite(
+            organization=organization,
+            email=invited_member.email,
+            user=request.user,
+            auth=request.auth,
+        ):
+            metrics.incr(
+                "member-invite.attempt",
+                instance="rate_limited",
+                skip_internal=True,
+                sample_rate=1.0,
+            )
+            return Response({"detail": ERR_RATE_LIMITED}, status=429)
+
+        if trigger_regenerate_token:
+            if request.access.has_scope("member:admin"):
+                with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+                    invited_member.regenerate_token()
+                    invited_member.save()
+            else:
+                return Response({"detail": ERR_INSUFFICIENT_SCOPE}, status=400)
```

Comment:
I think this should be 403 because they cannot perform this action due to insufficient scope

```suggestion
                raise PermissionDenied
```